2018/2/solution2.py

```diff
@@ -0,0 +1,28 @@
+def get_checksum(box_ids):
+    return (len([box_id for box_id in box_ids if contains_matching(2, box_id)])
+            * len([box_id for box_id in box_ids if contains_matching(3, box_id)]))
+
+
+def contains_matching(n, box_id):
+    letters = {c: box_id.count(c) for c in set(c for c in box_id)}
+    return any([True for letter in letters if letters[letter] == n])
+
+
+def get_common_letters(box_ids):
+    for i in range(0, len(box_ids)):
+        for j in range(i, len(box_ids)):
+            if differences(box_ids[i], box_ids[j]) == 1:
+                return remove_difference(box_ids[i], box_ids[j])
+    return None
```

Comment:
do you want to return None here, or throw an exception / assert, as we should never get here

Reply:
if I turned this into a list comprehension and returned the next element, that would remove the need. I returned None because pylint was complaining in this case.

Reply:
I was running into that too.  However, I think pytint is okay with an assert False or a raise Exception at the end of a function